tutorials/text/deduplication/semantic/semantic_step_by_step.ipynb

```diff
@@ -0,0 +1,1114 @@
+   "id": "8bec854a",
+   "metadata": {},
+   "source": [
+    "## Run Semantic Dedup workflow (w/o specifying `eps`)\n",
+    "\n",
+    "1. We intentionally don't specify `eps` so that we can show how to run `IdentifyDuplicates` as a separate stage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "e069a223",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2025-09-16 12:14:38,372\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
+      "2025-09-16 12:14:38,374\tINFO worker.py:1771 -- Connecting to existing Ray cluster at address: 127.0.1.1:6379...\n",
+      "2025-09-16 12:14:38,375\tINFO worker.py:1789 -- Calling ray.init() again after it has already been called.\n",
+      "2025-09-16 12:15:04,622\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
+      "2025-09-16 12:15:04,624\tINFO worker.py:1771 -- Connecting to existing Ray cluster at address: 127.0.1.1:6379...\n",
+      "2025-09-16 12:15:04,632\tINFO worker.py:1942 -- Connected to Ray cluster. View the dashboard at \u001b[1m\u001b[32mhttp://127.0.0.1:8265 \u001b[39m\u001b[22m\n",
+      "2025-09-16 12:15:04,647\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
+      "2025-09-16 12:15:04,649\tINFO worker.py:1771 -- Connecting to existing Ray cluster at address: 127.0.1.1:6379...\n",
+      "2025-09-16 12:15:04,649\tINFO worker.py:1789 -- Calling ray.init() again after it has already been called.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from nemo_curator.stages.deduplication.semantic import RankingStrategy, SemanticDeduplicationWorkflow\n",
+    "\n",
+    "semantic_workflow_path = os.path.join(cache_path, \"semantic_dedup\")\n",
+    "\n",
+    "workflow = SemanticDeduplicationWorkflow(\n",
+    "    input_path=embedding_output_path,\n",
+    "    output_path=semantic_workflow_path,\n",
+    "    n_clusters=100,\n",
+    "    # Since we use Id Generator in the embedding generation step, we need to specify the id field as `_curator_dedup_id`\n",
+    "    id_field=\"_curator_dedup_id\",\n",
+    "    embedding_field=\"embeddings\",\n",
+    "    ranking_strategy=RankingStrategy(metadata_cols=[\"cosine_dist_to_cent\"], ascending=True),\n",
+    "    # if eps is specified then it'll also run IdentifyDuplicates stage\n",
+    "    eps=None,\n",
+    ")\n",
+    "semantic_out = workflow.run()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a8743b9f",
+   "metadata": {},
+   "source": [
+    "#### KMeans Results\n",
+    "\n",
+    "1. `_curator_dedup_id` : The ids of the rows that belong to the cluster.\n",
+    "2. `embeddings` : These are later used for pairwise similarity\n",
+    "3. `l2_dist_to_cent` / `cosine_dist_to_cent` : This represents how far (l2 distance or cosine distance) is from our clusters centroid.\n",
```

Comment:
```suggestion
    "3. `l2_dist_to_cent` / `cosine_dist_to_cent` : This represents how far (l2 distance or cosine distance) a sample is from our cluster's centroid.\n",
```
It looks like some comments on the other tutorial can be carried over to this one too, thanks.